hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/atomic/RaftAtomicValueService.java

```diff
@@ -154,6 +154,9 @@ public int getAtomicValuesCount() {
         return atomicValues.size();
     }
 
+    // squid:S3824 ConcurrentHashMap.computeIfAbsent(K, Function<? super K, ? extends V>) locks the map, which *may* have an
+    // effect on throughput such that it's not a direct replacement
+    @SuppressWarnings("squid:S3824")
```

Comment:
if you look at the implementation they are simple object constructions which will not be blocking, hence I do not see why it may have an effect on throughput. Similarly for the other suppressed methods, it is similar situation.

Reply:
`newAtomicValue` is abstract, so you do not have guarantee that other implementations will not do something more complex

also the thread might get preempted when holding the lock and that might briefly block other threads for longer time than expected. Impact of this depends on usecase.

Reply:
> if you look at the implementation they are simple object constructions which will not be blocking, hence I do not see why it may have an effect on throughput. Similarly for the other suppressed methods, it is similar situation.

your probably right, but I thought it was safer to ignore this warning for any `ConcurrentHashMap` implementations.